### PR TITLE
[T2A2][W15-A1]How Si Wei

### PR DESCRIPTION
**Pull request by howsiwei**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -266,7 +266,7 @@ private static void processProgramArgs(String[] args) {
             setupGivenFileForStorage(args[0]);
         }
 
-        if(args.length == 0) {
+        if (args.length == 0) {
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:269]: good that you spotted this coding standard violation

Comment (mauris) [reply src/seedu/addressbook/AddressBook.java:269]: also noted that you have one change per commit. nice

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -485,14 +485,18 @@ private static String getMessageForPersonsDisplayedSummary(ArrayList<String[]> p
     private static ArrayList<String[]> getPersonsWithNameContainingAnyKeyword(Collection<String> keywords) {
         final ArrayList<String[]> matchedPersons = new ArrayList<>();
         for (String[] person : getAllPersonsInAddressBook()) {
-            final Set<String> wordsInName = new HashSet<>(splitByWhitespace(getNameFromPerson(person)));
-            if (!Collections.disjoint(wordsInName, keywords)) {
+            if (doesNameContainAnyKeyword(person, keywords)) {
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:488]: abstraction applied here helps readability. nice.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -1151,7 +1155,19 @@ private static String getUsageInfoForExitCommand() {
      * @return  string without the sign
      */
     private static String removePrefixSign(String s, String sign) {
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:1157]: as discussed in class, the term`sign` is not common (or in fact used elsewhere in this program) and we may want to just stick to the term `prefix` instead.

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -1151,7 +1155,19 @@ private static String getUsageInfoForExitCommand() {
...
-        return s.replace(sign, "");
+        return removePrefix(s, sign);
+    }
+    
+    /**
+    * Removes prefix from the given fullString if prefix occurs at the start of the string.
+    */
+    private static String removePrefix(String fullString, String prefix) {
+        if (!fullString.startsWith(prefix)) {
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:1165]: interesting check to ensure you replace the correct prefix.

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -1151,7 +1155,19 @@ private static String getUsageInfoForExitCommand() {
...
+            return fullString;
+        }
+        else {
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:1168]: coding standard violation though, `else` should stay in the same line as the previous closing brace.

**Comment by mauris** [comment]
Some comments added. Feel free to add more comment if you have any questions. Once you are done, you may close the PR after reading comments.

**Comment by howsiwei** [comment]
Thanks for the review

**Pull request closed without merging**